libyul/optimiser/FullInliner.cpp

```diff
@@ -178,8 +179,17 @@ bool FullInliner::shallInline(FunctionCall const& _funCall, YulString _callSite)
 		return false;
 
 	// Do not inline into already big functions.
-	if (m_functionSizes.at(_callSite) > 45)
-		return false;
+	if (
+		auto evmDialect = dynamic_cast<EVMDialect const*>(&m_dialect);
+		!evmDialect || !evmDialect->providesObjectAccess() || evmDialect->evmVersion() <= langutil::EVMVersion::homestead()
+	)
+	{
+		if (m_functionSizes.at(_callSite) > 45)
+			return false;
+	}
+	else
+		if (m_functionSizes.at(_callSite) > 350)
+			return false;
```

Comment:
By the way: this last offending test case does not work with old code generation at all, since it uses features we only implemented for Yul codegen - so there is no telling if this had even worked for old codegen :-).